### PR TITLE
alternative fix: the query engine is broken on some JVM because it relies on stack trace elements

### PR DESCRIPTION
**Pull request by pvojtechovsky**
This is an alternative solution to #1578

### CODE REVIEW

**Comment by pvojtechovsky** [comment]
Now there is solution, which detects whether detection based on CCE works on current JVM. 
If it works then future CCEs are avoided and client's CCEs are correctly rethrown.
If JVM is old or exotic, then it falls back to slow throwing of CCEs and to wrong ignoring of all exceptions.

File: src/main/java/spoon/reflect/visitor/chain/CtQueryImpl.java [hunk 1]
```diff
@@ -552,4 +533,61 @@ protected void handleResult(Object result, Object input) {
 			nextStep.accept(result);
 		}
 	}
+
+	private static Integer indexOfCallerInStack = null;
```
Comment (monperrus) [inline src/main/java/spoon/reflect/visitor/chain/CtQueryImpl.java]: remove mutability?

private final static Integer indexOfCallerInStack = getIndexOfCallerInStackOfLambda();

Comment (pvojtechovsky) [reply src/main/java/spoon/reflect/visitor/chain/CtQueryImpl.java]: ok, done

**Comment by monperrus** [comment]
thanks!

**Pull request merged**